pkg/operator/argo/argo.go

```diff
@@ -50,9 +50,9 @@ var runningStates = []string{
 	string(awfv1.NodeRunning),
 }
 
-func init() {
-	wfcs := wfclientset.NewForConfigOrDie(k8s.Config)
-	workflowClient = wfcs.ArgoprojV1alpha1().Workflows(cc.Namespace)
+func Init() {
```

Comment:
Is there anything Cortex-specific about this package? What do you think about passing in `namespace` and `restConfig` in `Init()` (is that all that's needed?), using it to populate a client, returning the client, and using the client downstream as a public variable in `config`, like with do with k8s and aws?

Now that I typed it out it sounds like it might be overkill, but I'll leave the comment here in case you want to consider it.

A lighter-weight option that still might be an improvement could be to pass in `namespace` and `restConfig` and save it in a package variable, and don't import the `config` package. The advantage would be a more explicit list of what's required to init argo, rather than relying on `config` to already be populated correctly. The disadvantage would be if we want changes in `config` to automatically populate to `argo`, which I could see. This would be less of an issue if we used a client as a global variable in `config`, since we would have just one place to update it.

The same discussion can be had regarding the `spark` package (but maybe not since `spark` probably has more dependencies on cortex-specific things).

What do you think? We can discuss it live.